sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala

```diff
@@ -264,12 +262,16 @@ private[hive] object HiveQl extends Logging {
    * Returns the HiveConf
    */
   private[this] def hiveConf: HiveConf = {
-    val ss = SessionState.get() // SessionState is lazy initialization, it can be null here
+    var ss = SessionState.get()
+    // SessionState is lazy initialization, it can be null here
     if (ss == null) {
-      new HiveConf()
-    } else {
-      ss.getConf
+      val original = Thread.currentThread().getContextClassLoader
+      val conf = new HiveConf(classOf[SessionState])
+      conf.setClassLoader(original)
+      ss = new SessionState(conf)
+      SessionState.start(ss)
```

Comment:
Actually, I guess we are fine. When we create a `HiveConf`, its internal classloader will be `Thread.currentThread().getContextClassLoader()`. So, we will not change the context classloader. But, it is better to have explicitly call `conf.setClassLoader` to set `original` as the classloader for this conf to avoid any confusion. What do you think?